docs/middleware.md

```diff
@@ -0,0 +1,76 @@
+# Using Next 12 middleware on Netlify
+
+Next 12 introduces a new feature called [middleware](https://nextjs.org/docs/middleware), which are functions that are
+run before a request has finished processing, and can be used to modify the request or replace the response. For
+example, it can change headers, rewrite the request path, or return a different response entirely.
+
+Next Middleware can run either in an edge function or at the origin. On Netlify, middleware runs at the origin as part
+of the normal Next.js server.
+
+## How to deploy Next 12 middleware
+
+Next 12 Middleware works out of the box with Netlify, and most functions will work unchanged. See
+[the middleware docs](https://nextjs.org/docs/middleware) for details of how to create them. There are however a few
+workarounds that are currently required for some features during the beta period:
```

Comment:
Minor edits (assuming these are Netlify functions in the first line)

```suggestion
Next 12 Middleware works out of the box with Netlify, and most Netlify functions will work unchanged. Refer to 
[the middleware docs](https://nextjs.org/docs/middleware) for details of how to create middleware functions. There are, however, a few
workarounds that are currently required for some features during the beta period:
```

Reply:
No, these are middleware functions.

Reply:
it may be good to specify that - for extreme clarity 😄